src/zenml/new/pipelines/pipeline.py

```diff
@@ -1108,7 +1108,17 @@ def _compile(
         integration_registry.activate_integrations()
 
         if config_path:
-            run_config = PipelineRunConfiguration.from_yaml(config_path)
+            with open(config_path, "r") as f:
+                from_config: Dict[str, Any] = yaml.load(
+                    f, Loader=yaml.SafeLoader
+                )
+            # pull out parameters relevant for PipelineRunConfiguration
+            from_config = {
```

Comment:
Does it make sense to only use the `steps` key here as the rest was already configured earlier?

Reply:
I just made it extensible, if in future we update the structure of the underlying models for any reason.

Reply:
But if I understand correctly, this now still applies the config twice right? Imagine the following:

```yaml
# config.yaml
extra:
  a: 1
```

```python
@pipeline
def my_pipeline():
  ...
  
p = my_pipeline.with_options(config_path="config.yaml")
p.configuration.extra["a"]  # 1

p.configure(extra={"a": 2})
p.configuration.extra["a"]  # 2

p()
# When running the pipeline, the config file gets reapplied and leads to (IMO) unexpected result of extra["a"] == 1
```

Reply:
I guess this also isn't ideal currently, as the config file only gets applied when running
```
...

p = my_pipeline.with_options(config_path="config.yaml")
p.configuration.extra["a"]  # KeyError
```

Reply:
I see, let me think about it 👍🏼

Reply:
Besides this, what about this approach to avoid double applying?
```python
        if config_path:
            with open(config_path, "r") as f:
                from_config: Dict[str, Any] = yaml.load(
                    f, Loader=yaml.SafeLoader
                )
            # pull out parameters relevant for PipelineRunConfiguration and not used in configure method
            configure_args = inspect.getfullargspec(self.configure)[0]
            from_config = {
                k: v
                for k, v in from_config.items()
                if k in PipelineRunConfiguration.__fields__
                and k not in configure_args
            }
            run_config = PipelineRunConfiguration(**from_config)
```

Reply:
> I guess this also isn't ideal currently, as the config file only gets applied when running
> 
> ```
> ...
> 
> p = my_pipeline.with_options(config_path="config.yaml")
> p.configuration.extra["a"]  # KeyError
> ```

What's inside your `config.yaml`?
This works for me:
```python
In [8]: p_ = p.with_options(config_path="tmp.yaml")

In [9]: p_.configuration
Out[9]: PipelineConfiguration(enable_cache=None, enable_artifact_metadata=None, enable_artifact_visualization=None, enable_step_logs=None, settings={}, extra={'a': 1}, failure_hook_source=None, success_hook_source=None, model_config_model=None, name='p')

In [10]: p.configuration
Out[10]: PipelineConfiguration(enable_cache=None, enable_artifact_metadata=None, enable_artifact_visualization=None, enable_step_logs=None, settings={}, extra={}, failure_hook_source=None, success_hook_source=None, model_config_model=None, name='p')
```
with
```yaml
extra:
  a: 1
```

Reply:
[8e96223](https://github.com/zenml-io/zenml/pull/1876/commits/8e96223721649f1789dce84b82c6aed7dd7651ed) should solve the double application issue while keeping it flexible for future extensions.

Reply:
Yep I like it. As for the example with the `KeyError`, I didn't actually run anything but assumed that's what was happening. However what I meant by "currently" was actually the code on `develop` and not the current state of your PR, so that was probably the misunderstanding